app/routes/company/components/CompaniesPage.tsx

```diff
@@ -84,31 +89,89 @@ const CompanyList = ({ companies = [] }: CompanyListProps) => (
   </div>
 );
 
+export const companiesDefaultQuery = {
+  search: '',
+  showInactive: 'false' as 'true' | 'false',
+};
+
+const filterCompanies = (
+  companies: ListCompany[],
+  query: typeof companiesDefaultQuery,
+) => {
+  return companies.filter((company) => {
+    const searchMatch =
+      !query.search ||
+      company.name.toLowerCase().includes(query.search.toLowerCase()) ||
+      company.description?.toLowerCase().includes(query.search.toLowerCase());
+
+    const activeMatch =
+      query.showInactive === 'true' || company.active !== false;
+
+    return searchMatch && activeMatch;
+  });
+};
+
 const CompaniesPage = () => {
   const [expanded, setExpanded] = useState(false);
+  const { query, setQueryValue } = useQuery(companiesDefaultQuery);
+
+  const unfilteredCompanies = useAppSelector(selectAllCompanies<ListCompany>);
+  const companies = useMemo(
+    () => filterCompanies(unfilteredCompanies, query),
+    [unfilteredCompanies, query],
+  );
```

Comment:
Delete these please, makes no sense to filter both in the frontend and in the backend